velox/exec/benchmarks/Crc32Benchmark.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/Random.h>
+#include <folly/hash/detail/ChecksumDetail.h>
+#include <iostream>
+#include <random>
+#include <vector>
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || \
+    defined(_M_IX86)
+#include <cpuid.h>
+#endif
+
+// Add the following definitions to allow Clion runs
+DEFINE_bool(gtest_color, false, "");
+DEFINE_string(gtest_filter, "*", "");
+
+namespace {
+
+bool isSSE42Supported() {
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || \
+    defined(_M_IX86)
+  unsigned int eax, ebx, ecx, edx;
+  __cpuid(1, eax, ebx, ecx, edx); // Get feature bits in ecx
+
+  // Bit 20 of ECX register indicates SSE4.2 support.
+  return (ecx & (1 << 20)) != 0;
+#else
+  return false; // Not an x86 architecture
+#endif
+}
+
+std::vector<uint8_t> generateRandomData(size_t size) {
+  std::vector<uint8_t> data(size);
+  std::mt19937_64 rng(folly::Random::rand64());
+  std::uniform_int_distribution<uint8_t> dist(0, 255);
+
+  for (size_t i = 0; i < size; ++i) {
+    data[i] = dist(rng);
+  }
+
+  return data;
+}
+
+void benchmarkCRC32Software(uint32_t dataSize) {
```

Comment:
Nit : Maybe I'm not looking hard enough, but seems like the same functions benchmarkCRC32Software( amd benchmarkCRC32Hardware are repeated twice. Can you confirm ?

Reply:
Ah... alright..Makes sense. 

So the goal here is to compare crc32 vs crc32c on different platforms. Do you have any early results to make a decision either way ?

Reply:
> Ah... alright..Makes sense.
> 
> So the goal here is to compare crc32 vs crc32c on different platforms. Do you have any early results to make a decision either way ?

In a day or two. I need to set up the Linux Fyre machine.